c10/core/TensorImpl.h

```diff
@@ -1830,6 +1830,9 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
   int64_t compute_numel() const {
     int64_t n = 1;
     for (auto s : sizes()) {
+      TORCH_CHECK(
```

Comment:
I'm concerned about the performance impact of changing this, since creating tensors so large seems niche. @ngimel and I didn't want to pursue a similar change previously. Maybe @swolchok would like to weigh-in now, too?

Reply:
We definitely shouldn't be adding this to `compute_numel` that's called for absolutely all new tensors. Having it in sparse utils is ok. see #43755

Reply:
@ngimel suggestion makes sense. When compute_numel overflows for dense tensors, something else will break anyway (unless one has a machine with 8 EiB RAM). But it is fairly easy to make numel to overflow for sparse tensors.

So, my plan is to implement safe compute_numel for sparse tensors and leave compute_numel for other tensors as it is.